django/db/models/fields/__init__.py

```diff
@@ -290,6 +293,30 @@ def _check_null_allowed_for_primary_keys(self):
     def _check_backend_specific_checks(self, **kwargs):
         return connection.validation.check_field(self, **kwargs)
 
+    def _check_migration_availability(self):
+        if self.removed_for_migrations:
+            return [
+                checks.Error(
+                    '%s has been removed except for support in historical '
```

Comment:
You are not using `removed_for_migrations.get('message')` as a primary source here.